enhancements/api-v2.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Quay API V2 design
+authors:
+  - "@sdadi"
+reviewers:
+  - "@syahmed"  
+  - "@sleesinc"
+  - "@obulatov"
+  - "@bcaton"
+approvers:
+  - "@syahmed"  
+  - "@sleesinc"
+  - "@obulatov"
+  - "@bcaton"
+creation-date: 2023-07-24
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# Quay API v2
+
+This is a proposal to design a generic strategy to improve Quay API's to support pagination, filtering and sorting.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+Currently, only a few APIs have the option to paginate results on the backend. The goal of the proposal is to have a re-usable system to perform actions on the backend and send the curated result on the frontend. Ideally frontend should only display the results and allow the backend to perform
+all the heavy lifting.
+
+## Motivation
+
+In the quay ui, filtering, sorting and occasionally paginating the results happen on the frontend. 
+This leads to high rendering time on the frontend and thereby a poor user experience.
```

Comment:
What are the implication of delegating that work to the server-side?
Can you add some implementation details w.r.t the database (query changes, indices, ...) involved,